src/material-experimental/form-field/testing/form-field-harness.ts

```diff
@@ -0,0 +1,226 @@
+import {MatInputHarness} from '@angular/material-experimental/input/testing';
+import {MatSelectHarness} from '@angular/material-experimental/select/testing';
+import {FormFieldHarnessFilters} from './form-field-harness-filters';
+
+// TODO(devversion): support datepicker harness once developed (COMP-203).
+// Also support chip list harness.
+/** Possible harnesses of controls which can be bound to a form-field. */
+export type FormFieldControlHarness = MatInputHarness|MatSelectHarness;
+
+/**
+ * Base class for custom form-field control harnesses. Harnesses for
+ * custom controls with form-fields need to implement this interface.
+ */
+export interface CustomFormFieldControlHarness extends ComponentHarness {}
+
+/**
+ * Harness for interacting with a standard Material form-field's in tests.
+ * @dynamic
+ */
+export class MatFormFieldHarness extends ComponentHarness {
+  static hostSelector = '.mat-form-field';
+
+  /**
+   * Gets a `HarnessPredicate` that can be used to search for an form-field with
+   * specific attributes.
+   * @param options Options for narrowing the search:
+   *   - `selector` finds a form-field that matches the given selector.
+   * @return a `HarnessPredicate` configured with the given options.
+   */
+  static with(options: FormFieldHarnessFilters = {}): HarnessPredicate<MatFormFieldHarness> {
+    return new HarnessPredicate(MatFormFieldHarness, options);
+  }
+
+  private _prefixContainer = this.locatorForOptional('.mat-form-field-prefix');
+  private _suffixContainer = this.locatorForOptional('.mat-form-field-suffix');
+  private _label = this.locatorForOptional('.mat-form-field-label');
+  private _errors = this.locatorForAll('.mat-error');
+  private _hints = this.locatorForAll('mat-hint,.mat-hint');
+
+  private _inputControl = this.locatorForOptional(MatInputHarness);
+  private _selectControl = this.locatorForOptional(MatSelectHarness);
+
+  /** Gets the appearance of the form-field. */
+  async getAppearance(): Promise<'legacy'|'standard'|'fill'|'outline'> {
+    const hostClasses = await (await this.host()).getAttribute('class');
+    if (hostClasses !== null) {
+      const appearanceMatch =
+          hostClasses.match(/mat-form-field-appearance-(legacy|standard|fill|outline)(?:$| )/);
+      if (appearanceMatch) {
+        return appearanceMatch[1] as 'legacy' | 'standard' | 'fill' | 'outline';
+      }
+    }
+    throw Error('Could not determine appearance of form-field.');
+  }
+
+  /**
+   * Gets the harness of the control that is bound to the form-field. Only
+   * default controls such as "MatInputHarness" and "MatSelectHarness" are
+   * supported.
+   */
+  async getControl(): Promise<FormFieldControlHarness|null>;
+
+  /**
+   * Gets the harness of the control that is bound to the form-field. Searches
+   * for a control that matches the specified harness type.
+   */
+  async getControl<X extends CustomFormFieldControlHarness>(type: ComponentHarnessConstructor<X>):
```

Comment:
Lets make this able to take a `HarnessPredicate` as well, that way people can do something like `MatInputHarness.with({selector: '#ff-control'})`

Reply:
Hmm. I'm not sure about that. My thinking is that if they want to search for specific harnesses anyway, they don't need to go through the harness anyway.

Reply:
Done as of Slack discussion.